samples/copyclean.py

```diff
@@ -0,0 +1,129 @@
+#   Copyright 2015 Christian Nilsson <nikize@gmail.com>
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import print_function
+
+import sys
+
+from Registry import RegistryParse
+import struct
+
+class Copy(RegistryParse.REGFBlock):
+    """
+    Parsing of file as a Windows Registry file.
+    """
+    def __init__(self, filelikeobject, writer):
+        """
+        Constructor.
+        Arguments:
+        - `filelikeobject`: A file-like object with a .read() method.
+              If a Python string is passed, it is interpreted as a filename,
+              and the corresponding file is opened.
+        """
+        try:
+            self._buf = filelikeobject.read()
+        except AttributeError:
+            with open(filelikeobject, "rb") as f:
+                self._buf = f.read()
+        self._buflen = len(self._buf)
+        
+        self._writer = writer
+
+        super(Copy, self).__init__(self._buf, 0, False)
+
+    def pack_dword(self, offset, *data):
+        """
+        write little-endian DWORDs (4 bytes) to the relative offset.
+        Arguments:
+        - `offset`: The relative offset from the start of the block.
+        - `data`: The data to be writen, can be multiple
+        """
+        print("write dword 0x%08x at offset 0x%08x" % (data[0], self._offset + offset) )
+        #return struct.pack_into(str("<I"), self._writer, self._offset + offset, *data)
+        self._writer.seek(offset)
+        self._writer.write(struct.pack(str("<I"), *data))
+
+    def pack_qword(self, offset, *data):
+        """
+        write little-endian QWORDs (8 bytes) to the relative offset.
+        Arguments:
+        - `offset`: The relative offset from the start of the block.
+        - `data`: The data to be writen, can be multiple
+        """
+        print("write qword 0x%016x at offset 0x%08x" % (data[0], self._offset + offset) )
+        self._writer.seek(offset)
+        self._writer.write(struct.pack(str("<Q"), *data))
+
+
+
+    def copy(self):
+        idx = 0x0
+        xsum = 0
+        #Read header including name, ignore junk (or atleast unknown)
+        while idx < 0x30+64:
+            dwd = self.unpack_dword(idx)
+            xsum ^= dwd
+            #read and write all as dword even if some are string.
+            self.pack_dword(idx, dwd) 
+            idx += 0x4
+        #write checksum to file
+        self.pack_dword(0x1FC, xsum)
+        
+        for hbin in self.hbins():
```

Comment:
It will not keep "junk" and will calculate a valid checksum, don't know how picky windows is, but it works fine loading the hive after cleanup etc.
Suggestion on needed documentation? how about replacing L80 with
`#Read header including name, skipping unused data and recalculate checksum.`
And L84: `#copy fields as dwords for simple checksum calculation, even if some parts are strings.`